packages/testing/src/execution_testing/cli/pytest_commands/plugins/shared/benchmarking.py

```diff
@@ -110,14 +110,30 @@ def get_opcode_counts_for_test(
 def pytest_collection_modifyitems(
     config: pytest.Config, items: list[pytest.Item]
 ) -> None:
-    """Filter tests based on repricing marker when benchmark options are used."""
+    """Filter tests based on repricing marker."""
     gas_benchmark_value = config.getoption("gas_benchmark_value")
     fixed_opcode_count = config.getoption("fixed_opcode_count")
 
     # Only filter if either benchmark option is provided
     if not gas_benchmark_value and fixed_opcode_count is None:
         return
 
+    # In --fixed-opcode-count mode, we only support tests that meet all of the following:
+    #   - The test uses the benchmark_test fixture
+    #   - The benchmark test uses a code generator
+    #
+    # Here we filter out tests that do not use the benchmark_test fixture.
+    # Note: At this stage we cannot filter based on whether a code generator is used.
+    if fixed_opcode_count or fixed_opcode_count == "":
```

Comment:
nit: I think this can be simplified to:

```suggestion
    if fixed_opcode_count is not None:
```

right?